src/workerd/jsg/modules.h

```diff
@@ -175,7 +175,18 @@ class ModuleRegistry {
                v8::Local<v8::Module> module,
                kj::Maybe<SyntheticModuleInfo> maybeSynthetic = nullptr);
 
-    ModuleInfo(jsg::Lock& js, kj::StringPtr name, kj::ArrayPtr<const char> content);
+    enum class CompileFlags {
+      NONE,
+      EXTERNAL,
+      // The EXTERNAL flag tells the compile operation to treat the content as an external
+      // string wrapping an immutable buffer outside the V8 heap. When this flag is not set,
+      // the content is copied into the V8 heap.
+    };
```

Comment:
Side note: I anticipate adding an additional flag here to control use of a compile cache. That'll come separately, however.